webapp/src/toolboxeditor.tsx

```diff
@@ -151,6 +154,25 @@ export abstract class ToolboxEditor extends srceditor.Editor {
             }).filter(subns => !!subns);
         }
 
+        function isExtension(ns: string, md: pxtc.CommentAttrs) {
+            const nsAttr = getBlocksEditor().getNamespaceAttrs(ns);
+
+            const foundExtension = pkg.mainEditorPkg().pkgAndDeps().find(p => {
+                const ext = p.getKsPkg()
+                if (!ext) return false
+                const namespaces = pkg.getNsForPkg(ext.id)
+                if (!namespaces) return false
+                return namespaces.indexOf(ns) >= 0
+            })
+            const extensionPkg = foundExtension?.getKsPkg();
+
+            let trgConfigFetch = data.getDataWithStatus("target-config:");
+            let trgConfig = trgConfigFetch.data as pxt.TargetConfig;
+
+            const hasDel = (nsAttr._def && nsAttr._def.parts.length > 0) || (foundExtension && extensionPkg.id != "core" && !trgConfig.packages.extensionsToIgnore.includes(extensionPkg.id))
```

Comment:
do we need to account for ---{variant}s here? e.g. when you select hardware to download and it swaps over to hardware variant pkgs https://github.com/microsoft/pxt-common-packages/blob/master/libs/core---stm32/pxt.json#L2